pyccel/codegen/printing/fcode.py

```diff
@@ -194,29 +194,19 @@ class FCodePrinter(CodePrinter):
     }
 
 
-    def __init__(self, parser, prefix_module = None):
+    def __init__(self, filename, prefix_module = None):
```

Comment:
This printer class has undergone quite a lot of changes. Could you please summarize them for me?

Reply:
There are not as many changes as it may seem. They are:
- Add a scope/namespace to the `CodePrinter` and ensure that it is correctly set at each entry/exit of a `_print_ScopedNode` function
- Replace all expressions of the type `self.parser.*` with the new equivalent `self.namespace.*`
- Use new `expand_to_loops` prototype

Reply:
There are slightly more changes in `codegen/printing/ccode.py` where the `_additional_declare` list has also been removed so that new variables are handled in the same way as in `codegen/printing/fcode.py` (ie. via the Scope)

Reply:
All is clear now, thanks